CC/Sounder/Radio.cc

```diff
@@ -6,33 +6,61 @@
 void Radio::dev_init(Config* _cfg, int ch, double rxgain, double txgain)
 {
     // these params are sufficient to set before DC offset and IQ imbalance calibration
-    dev->setAntenna(SOAPY_SDR_RX, ch, "TRX");
-    dev->setBandwidth(SOAPY_SDR_RX, ch, _cfg->bwFilter);
-    dev->setBandwidth(SOAPY_SDR_TX, ch, _cfg->bwFilter);
+    if (!kUseUHD) {
+        dev->setAntenna(SOAPY_SDR_RX, ch, "TRX");
+        dev->setBandwidth(SOAPY_SDR_RX, ch, _cfg->bwFilter);
+        dev->setBandwidth(SOAPY_SDR_TX, ch, _cfg->bwFilter);
+        dev->setFrequency(SOAPY_SDR_RX, ch, "RF", _cfg->radioRfFreq);
+        dev->setFrequency(SOAPY_SDR_RX, ch, "BB", _cfg->nco);
+        dev->setFrequency(SOAPY_SDR_TX, ch, "RF", _cfg->radioRfFreq);
+        dev->setFrequency(SOAPY_SDR_TX, ch, "BB", _cfg->nco);
+    } else {
+        std::cout << std::endl
+                  << "Init USRP channel: " << ch << std::endl;
+        dev->setAntenna(SOAPY_SDR_TX, ch, "TX/RX");
+        dev->setAntenna(SOAPY_SDR_RX, ch, "RX2"); // or "TX/RX"
+        std::cout << "Tx antenna: " << dev->getAntenna(SOAPY_SDR_TX, ch)
+                  << ", Rx antenna: " << dev->getAntenna(SOAPY_SDR_RX, ch) << std::endl;
```

Comment:
Can we move this print out of this if and make it work for both cases?

Reply:
Done